### DISCUSSION TITLE
MDX files and Internationalization (i18n)

### DISCUSSION THREAD

**Discussion by figueroaignacio** [vercel/next.js#68374] [Help]
### Summary

Hello everyone! I'm having a question about internationalization with mdx files.

I'm creating a website with Next.js and Velite (contentlayer alternative), where I write posts and guides about web development and I want it to be in two languages so far (Spanish and English).

Is there any way to traduce the mdx files?

If yes, how?

**Comment by Juneezee** [comment]
A good starting point is to use the [next-intl](https://github.com/amannn/next-intl) package. You can read their docs page here: https://next-intl-docs.vercel.app/docs/environments/mdx

**Reply by figueroaignacio** [reply]
Heey, thanks for your answer.

I'm using a contentaler alternative (Velite) that actually works the same.

With this I had to create the collections and I had to create a /content folder where there's the /posts folder with its respectives mdx files.

What should I make to translate those mdx files from the /content/posts folder? That's my main question.

And yeah I'm using next-intl but I don't know how to actually translate the mdx files so if you can help me that would be great.

Cheers 🚀

**Reply by Juneezee** [reply]
If you are using App Router, then you can follow https://nextjs.org/docs/app/building-your-application/routing/internationalization

This is also shown in the first code block on https://next-intl-docs.vercel.app/docs/environments/mdx, where you have a `[locale]` (or `[lang]` etc.) dynamic route [^1]:

```
src
└── app
    └── [locale]
        ├── page.tsx
        ├── en.mdx
        └── de.mdx
```

You can read the value of `[locale]` through `params` [^2] in `page.tsx` and load either `en.mdx` or `de.mdx`.

[^1]: https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#convention
[^2]: https://nextjs.org/docs/app/api-reference/file-conventions/page#params-optional

**Reply by sanne-v** [reply]
Documentation of next-intl seems to be behind a pay wall now 😓 

https://next-intl-docs.vercel.app/docs/environments/mdx

**Reply by sverrirarnors** [reply]
@sanne-v you can retrieve an old version from the `next-intl` repo: https://github.com/amannn/next-intl/blob/7c3a6a16897212291232e934a9889048f809b791/docs/pages/docs/environments/mdx.mdx
